src/components/pages/Settings/ArNSSettings.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  AOProcess,
+  IO,
+  IO_DEVNET_PROCESS_ID,
+  IO_TESTNET_PROCESS_ID,
+} from '@ar.io/sdk';
+import Toggle from '@src/components/inputs/toggle/Toggle';
+import ArweaveID, {
+  ArweaveIdTypes,
+} from '@src/components/layout/ArweaveID/ArweaveID';
+import { ArweaveCompositeDataProvider } from '@src/services/arweave/ArweaveCompositeDataProvider';
+import { ArweaveTransactionID } from '@src/services/arweave/ArweaveTransactionID';
+import { SimpleArweaveDataProvider } from '@src/services/arweave/SimpleArweaveDataProvider';
+import { useGlobalState, useWalletState } from '@src/state';
+import { isArweaveTransactionID } from '@src/utils';
+import { IO_PROCESS_ID } from '@src/utils/constants';
+import { Input } from 'antd';
+import Arweave from 'arweave';
+import { RotateCcw } from 'lucide-react';
+import { useEffect, useState } from 'react';
+
+import './styles.css';
+
+function ArNSSettings() {
+  const [{ ioProcessId, aoClient, gateway }, dispatchGlobalState] =
+    useGlobalState();
+  const [{ wallet }] = useWalletState();
+  const [registryAddress, setRegistryAddress] = useState<string>(
+    ioProcessId?.toString(),
+  );
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(true);
+
+  useEffect(() => {
+    setRegistryAddress(ioProcessId?.toString());
+  }, [ioProcessId]);
+
+  function confirmSetting(id: string) {
+    if (isArweaveTransactionID(id)) {
+      dispatchGlobalState({
+        type: 'setIoProcessId',
+        payload: id.trim(),
+      });
+
+      const arIOContract = IO.init({
+        process: new AOProcess({
+          processId: id.trim(),
+          ao: aoClient,
+        }),
+        ...(wallet?.arconnectSigner ? { signer: wallet.arconnectSigner } : {}),
+      });
+      dispatchGlobalState({
+        type: 'setArIOContract',
+        payload: arIOContract,
+      });
+
+      const arweave = new Arweave({
+        host: gateway,
+        protocol: 'https',
+      });
+      const arweaveDataProvider = new SimpleArweaveDataProvider(arweave);
+
+      const provider = new ArweaveCompositeDataProvider({
+        contract: arIOContract,
+        arweave: arweaveDataProvider,
+      });
+
+      dispatchGlobalState({
+        type: 'setGateway',
+        payload: { gateway, provider },
+      });
+    }
+  }
+
+  const labelClass =
```

Comment:
I generally find it easier to see the layout when the tailwind classes are inline with the elemtns.

Reply:
You can see it in the code when you hover over the variable - not as friendly for reviewing but its overall better DX for me.